commands/scale_cluster.go

```diff
@@ -38,115 +39,307 @@ Examples:
-func scaleClusterPreRunOutput(cmd *cobra.Command, cmdLineArgs []string) {
-	args := defaultScaleClusterArguments()
-	err := verifyScaleClusterPreconditions(args, cmdLineArgs)
+		return nil, microerror.Mask(err)
+	}
 
-	if err == nil {
-		return
+	// We have to marshal and unmarshal here because the generated client gives us
+	// a map[string]interface and we want to unmarshal it into the actual
+	// apiextensions type.
+	m, err := json.Marshal(response.Payload)
+	if err != nil {
+		return nil, err
 	}
 
-	handleCommonErrors(err)
+	var status v1alpha1.StatusCluster
+
+	err = json.Unmarshal(m, &status)
+	if err != nil {
+		return nil, err
+	}
 
-	// print non-common error
-	fmt.Println(color.RedString(err.Error()))
-	os.Exit(1)
+	return &status, nil
 }
 
-// verifyScaleClusterPreconditions does a few general checks and returns an error in case something is missing.
-func verifyScaleClusterPreconditions(args scaleClusterArguments, cmdLineArgs []string) error {
-	if config.Config.Token == "" && args.authToken == "" {
-		return microerror.Mask(notLoggedInError)
+// scaleCluster is the actual function submitting the API call and handling the response.
+func scaleCluster(args scaleClusterArguments) error {
+	// Preparing API call.
+	reqBody := &models.V4ModifyClusterRequest{
+		Scaling: &models.V4ModifyClusterRequestScaling{
+			Max: args.workersMax,
+			Min: args.workersMin,
+		},
 	}
-	if len(cmdLineArgs) == 0 {
-		return microerror.Mask(clusterIDMissingError)
+
+	// perform API call
+	if args.verbose {
+		fmt.Println("Sending API request to modify cluster")
+	}
+
+	auxParams := ClientV2.DefaultAuxiliaryParams()
+	auxParams.ActivityName = scaleClusterActivityName
+
+	_, err := ClientV2.ModifyCluster(args.clusterID, reqBody, auxParams)
+	if err != nil {
+		if IsAPIError(err) {
+			clientErr, ok := err.(*clienterror.APIError)
+			if ok {
+				switch clientErr.HTTPStatusCode {
+				case http.StatusForbidden:
+					return microerror.Mask(accessForbiddenError)
+				case http.StatusUnauthorized:
+					return microerror.Mask(notAuthorizedError)
+				case http.StatusNotFound:
+					return microerror.Mask(clusterNotFoundError)
+				case http.StatusInternalServerError:
+					return microerror.Mask(internalServerError)
+				}
+			}
+		}
+
+		return microerror.Mask(err)
 	}
+
 	return nil
 }
 
 // scaleClusterRunOutput invokes the actual cluster scaling and prints the result and/or errors.
 func scaleClusterRunOutput(cmd *cobra.Command, cmdLineArgs []string) {
-	args := defaultScaleClusterArguments()
-	args.clusterID = cmdLineArgs[0]
+	if len(cmdLineArgs) == 0 {
+		handleCommonErrors(clusterIDMissingError)
+	}
+	// Get all necessary information for defaulting and confirmation.
+	clusterDetails, err := getClusterDetails(cmdLineArgs[0], scaleClusterActivityName)
+	if err != nil {
+		fmt.Println(color.RedString("Error getting cluster details!"))
+		handleCommonErrors(err)
+		fmt.Println(color.RedString(err.Error()))
+		os.Exit(1)
```

Comment:
We should start wrapping functionality and have proper error handling. The caller should then be the only place where the program is killed. Having this all over the place is super messy. We should aim for one line error handling. Not four. We would then also improve maintainability and testability.